packages/application-generic/src/usecases/store-subscriber-jobs/store-subscriber-jobs.usecase.ts

```diff
@@ -53,24 +44,41 @@ export class StoreSubscriberJobs {
         return list;
       }, []);
 
-    await this.notificationRepository.update(
-      {
-        _organizationId: firstJob._organizationId,
-        _id: firstJob._notificationId,
-      },
-      {
-        $set: {
-          channels: channels,
+    await Promise.all([
+      this.notificationRepository.update(
+        {
+          _organizationId: firstJob._organizationId,
+          _id: firstJob._notificationId,
         },
-      }
-    );
+        {
+          $set: {
+            channels: channels,
+          },
+        }
+      ),
+      this.addJob.execute({
+        userId: firstJob._userId,
+        environmentId: firstJob._environmentId,
+        organizationId: firstJob._organizationId,
+        jobId: firstJob._id,
+        job: firstJob,
+      }),
+    ]);
+  }
 
-    await this.addJob.execute({
-      userId: firstJob._userId,
-      environmentId: firstJob._environmentId,
-      organizationId: firstJob._organizationId,
-      jobId: firstJob._id,
-      job: firstJob,
-    });
+  @Instrument()
+  private createJobsExecutionDetails(storedJobs: JobEntity[]) {
+    for (const job of storedJobs) {
+      this.createExecutionDetails.execute(
+        CreateExecutionDetailsCommand.create({
+          ...CreateExecutionDetailsCommand.getDetailsFromJob(job),
+          detail: DetailEnum.STEP_CREATED,
+          source: ExecutionDetailsSourceEnum.INTERNAL,
+          status: ExecutionDetailsStatusEnum.PENDING,
+          isTest: false,
+          isRetry: false,
+        })
+      );
+    }
```

Comment:
Wonder if we can do a bulk write here in the future

Reply:
I already did it

Reply:
I want to push it in a different PR, and also thats why I added the Instrument here, I want to have a more precise understanding of the affect